src/js/controllers/clipboard_controller.js

```diff
@@ -0,0 +1,53 @@
+import { Controller } from '@hotwired/stimulus'
+
+export default class extends Controller {
+  static targets = ['source', 'ctaText', 'successText']
+  static classes = ['toggleSuccess']
+  static values = {
+    successDelay: {
+      default: 1000,
+      type: Number,
+    },
+  }
+
+  successTimeout = null
+
+  copy() {
+    if (this.hasSourceTarget) {
```

Comment:
NIT: I think an early returns is cleaner in this case. 

Especially in this case when all the code has to be indented. 

```
if (!this.hasSourceTarget) {
    return
}
```

Reply:
totes 👍